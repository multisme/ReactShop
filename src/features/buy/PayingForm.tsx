import React, { useState } from "react";
import {useSelector} from "react-redux";

import { Elements, CardElement } from "@stripe/react-stripe-js";
import { loadStripe } from "@stripe/stripe-js";

import {cartSelector} from "features/cart/cartSlice";
import StripeCheckoutForm from "features/buy/StripeCheckoutForm";
const stripePromise = loadStripe("pk_test_JJ1eMdKN0Hp4UFJ6kWXWO4ix00jtXzq5XG");


const PayingForm = () => {
  const price = useSelector(cartSelector);
 /* if (price == 0){
          return <h3>Please buy some stuff before Paying</h3>
          }
          */
  return (
    <div id="payingForm" className="rectangle green-border">
    <Elements stripe={stripePromise}>
    <StripeCheckoutForm total={price}/>
    </Elements>
    </div>
  );
};

export default PayingForm;
