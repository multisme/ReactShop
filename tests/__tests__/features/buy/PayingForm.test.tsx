import PayingForm from "features/buy/PayingForm";
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from "react-router-dom";
import * as redux from "react-redux";
import { Provider } from "react-redux";
import { mount } from "enzyme";

const mockStore = configureStore();

describe("PayingFrom", () => {
                let store: any;

                beforeEach(() => {
                        store = mockStore({ });
                        const mock = jest.spyOn(redux, 'useSelector');
                        mock.mockReturnValue(0)
                })
        it("renders", () => {

                const component = mount(
                <Router>
                        <Provider store={store}>
                                <PayingForm />
                        </Provider>
                </Router>
                );
                expect(component).toMatchSnapshot();
        })
})
