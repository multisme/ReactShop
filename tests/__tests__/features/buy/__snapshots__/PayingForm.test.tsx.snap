// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayingFrom renders 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <PayingForm>
        <div
          className="rectangle"
          id="payingForm"
        >
          <Elements
            stripe={Promise {}}
          >
            <StripeCheckoutForm
              total={0}
            >
              <form
                onSubmit={[Function]}
              >
                <CardElement
                  options={
                    Object {
                      "style": Object {},
                    }
                  }
                >
                  <div />
                </CardElement>
                <Link
                  to="/thank"
                >
                  <LinkAnchor
                    href="/thank"
                    navigate={[Function]}
                  >
                    <a
                      href="/thank"
                      onClick={[Function]}
                    >
                      <button
                        disabled={true}
                        type="submit"
                      >
                        Pay 
                        0
                        â‚¬
                      </button>
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  to="/home"
                >
                  <LinkAnchor
                    href="/home"
                    navigate={[Function]}
                  >
                    <a
                      href="/home"
                      onClick={[Function]}
                    >
                      <button>
                        CANCEL
                      </button>
                    </a>
                  </LinkAnchor>
                </Link>
              </form>
            </StripeCheckoutForm>
          </Elements>
        </div>
      </PayingForm>
    </Provider>
  </Router>
</BrowserRouter>
`;
